### PR TITLE
simplify alloc impl, make click to source work locally

### PR DESCRIPTION
**Pull request by jakebolewski**
also write out a CSV for machine readable artifacts

### CODE REVIEW

**Comment by jakebolewski** [comment]
the csv is nice because you can then download it and view it in vscode and click to source works

**Comment by jakebolewski** [comment]
bors try

File: perf/allocs.jl [hunk 1]
```diff
@@ -149,65 +150,33 @@ function plot_allocs(
...
-        Plots.plot!(
-            [0],
-            [bytes];
-            seriestype = :scatter,
-            label = label,
-            markershape = markershape[1],
-            markersize = 1 + bytes / max_bytes * 10,
-        )
-        markershape = (markershape[end], markershape[1:(end - 1)]...)
+    all_kbytes = map(b -> div(b, 10^3), all_bytes)
```
Comment (charleskawczynski) [inline perf/allocs.jl:172]:
```suggestion
    all_kbytes = map(b -> div(b, 1024), all_bytes) # convert to KiB
```
Maybe we should report with KiB, to be consistent with BenchmarkTools?

**Comment by charleskawczynski** [comment]
I just finished setting it up with PrettyTables:

```
[ Info: Allocations for MA_example
[ Info: Number of sampled allocating sites: 4
[ Info: Number of unique allocating sites: 2
┌───────────────────┬─────────────┬────────────────────────────────────────────────────────────┐
│     Allocations % │ Allocations │                                     src code / line number │
│ (alloc_i/∑allocs) │       (KiB) │                                                            │
├───────────────────┼─────────────┼────────────────────────────────────────────────────────────┤
│          0.606143 │   468718.75 │ MonitorAllocations.jl/test/rep_workload.jl, line number: 9 │
│          0.393857 │    304562.5 │ MonitorAllocations.jl/test/rep_workload.jl, line number: 8 │
└───────────────────┴─────────────┴────────────────────────────────────────────────────────────┘
```
Both sound good to me, I really like the idea of something more copy/paste-friendly. I would suggest we add this percentage column to the table, so that it's immediately clear what percentage each contribution is compared to the whole distribution.

**Comment by jakebolewski** [comment]
bors try

**Comment by jakebolewski** [comment]
@charleskawczynski has created a new package for this incorporating some of these ideas: https://github.com/CliMA/ReportAllocations.jl

**Comment by jakebolewski** [comment]
Superseded by #441

**Pull request closed without merging**